src/microbit/test/test_image.py

```diff
@@ -0,0 +1,278 @@
+                Image("999:999:999:000:"),
+                Image("999:999:999:000:"),
+            ),
+        ],
+    )
+    def test_add(self, target, value, actual):
+        result = target + value
+        assert result._Image__LED == actual._Image__LED
+
+    @pytest.mark.parametrize(
+        "target, value, err_message",
+        [
+            (
+                Image("012:345:678:900:"),
+                2,
+                CONSTANTS.UNSUPPORTED_ADD_TYPE + f"'{type(Image())}', '{type(2)}'",
+            ),
+            (
+                Image("012:345:678:900:"),
+                [],
+                CONSTANTS.UNSUPPORTED_ADD_TYPE + f"'{type(Image())}', '{type([])}'",
+            ),
+        ],
+    )
+    def test_add_typeerror(self, target, value, err_message):
+        with pytest.raises(TypeError, match=err_message):
+            target + value
+
+    # ADD - VALUEERROR
+    @pytest.mark.parametrize(
+        "target, value", [(Image(2, 3), Image(3, 3)), (Image(2, 1), Image(0, 0))]
+    )
+    def test_add_valueerror(self, target, value):
+        with pytest.raises(ValueError, match=CONSTANTS.SAME_SIZE_ERR):
+            target + value
+
+    @pytest.mark.parametrize(
+        "initial, actual",
+        [
+            (Image("0:000:00:0000:"), Image("0000:0000:0000:0000:")),
+            (Image("12125:1212:12:1:"), Image("12125:12120:12000:10000:")),
+        ],
+    )
+    def test_uneven_strings(self, initial, actual):
+        assert initial._Image__LED == actual._Image__LED
+
+    @pytest.mark.parametrize(
+        "image, repr_actual, str_actual",
+        [
+            (
+                Image("05150:05050:05050:99999:09990:"),
+                "Image('05150:05050:05050:99999:09990:')",
+                "Image('\n	05150:\n	05050:\n	05050:\n	99999:\n	09990:\n')",
+            ),
+            (Image(""), "Image('')", "Image('\n')"),
+            (
+                Image("00000:00000:00000:00000:00000:"),
+                "Image('00000:00000:00000:00000:00000:')",
+                "Image('\n	00000:\n	00000:\n	00000:\n	00000:\n	00000:\n')",
+            ),
+            (
+                Image("00:00:00:00:"),
+                "Image('00:00:00:00:')",
+                "Image('\n	00:\n	00:\n	00:\n	00:\n')",
+            ),
+        ],
+    )
+    def test_str(self, image, repr_actual, str_actual):
+        repr_output = repr(image)
+        str_output = str(image)
+        assert repr_actual == repr_output
+        assert str_actual == str_output
+
+    # @pytest.mark.parametrize(
+    #     "initial_img_string, actual",
+    #     [
+    #         ("0:000:00:0000:", Image("0000:0000:0000:0000:")),
+    #         ("0000\n0000\n0000\n0000\n", Image("0000:0000:0000:0000:")),
+    #         ("0\n000\n00\n0000\n", Image("0000:0000:0000:0000:")),
+    #     ],
+    # )
+    # def test_string_processing_constructor(self, initial_img_string, actual):
+    #     assert Image(initial_img_string)._Image__LED == actual._Image__LED

```

Comment:
Perhaps remove this comment. If we want to include this code in the future, the author can find it on git.

Reply:
oops, didn't mean to leave that in! thanks :)